Things I had laying around (June 2025 edition) (#507)
the1lab/1lab: src/Homotopy/Space/Rose.lagda.md

Reviewer: This page is so pretty 🌹

<!--
```agda
open import 1Lab.Path.Reasoning
open import 1Lab.Prelude

open import Algebra.Group

open import Data.Sum.Properties
open import Data.List.Base
open import Data.Dec.Base
open import Data.Sum.Base

import Algebra.Group.Free.Words as W
import Algebra.Group.Cayley as C
```
-->

```agda
module Homotopy.Space.Rose where
```

# Roses

The **rose with $n$ petals** $\bigvee_n S^1$ is the space described by
the higher inductive type containing a single point `base`{.Agda} and
$n$ distinct path constructors from `base`{.Agda} to `base`{.Agda}. For
example, the following picture illustrates the rose with 3 petals.

~~~{.quiver}
\begin{tikzpicture}[scale=0.625]

\begin{polaraxis}[grid=none, axis lines=none]
  \addplot[mark=none,domain=0:360,samples=300, thick] {cos(180 + x * 3)};
  \node[draw,circle,label=right:{\Large $\rm{base}$},fill,outer sep=0.2cm, inner sep=0pt, minimum size=0.2cm] (a0) at (0, 0) {};
\end{polaraxis}

\end{tikzpicture}
~~~

More generally, we can allow the arity $n$ to be given by an arbitrary

> type $A$--- but for the computation of the fundamental group below, we

Reviewer: ```suggestion
type $A$ --- but for the computation of the fundamental group below, we
```

Author: nah

Reviewer: Doesn't this look a bit weird?

![TOs-0R](https://github.com/user-attachments/assets/844251ec-ebad-4dda-b7ba-1a5345cf91f4)

Author: this is how i've always written my em dashes

Author: i don't live in chicago

  \arrow[""{name=0, anchor=center, inner sep=0}, "{\rm{loop}^{x \cdot w}}"', from=2-2, to=4-2]
  \arrow["{\rm{refl}}"', from=4-2, to=4-4]
  \arrow[""{name=1, anchor=center, inner sep=0}, "{\rm{loop}^w}", from=2-4, to=4-4]
  \arrow["{\rm{refl}}"{description}, dashed, from=1-1, to=1-5]
  \arrow["{\rm{loop}^w}"', dashed, from=1-1, to=5-1]
  \arrow["{\rm{loop}\ x}"{description}, dashed, from=5-1, to=5-5]
  \arrow["{\rm{loop}^w}", dashed, from=1-5, to=5-5]
  \arrow["{\rm{refl}}"{pos=0.2}, from=2-2, to=1-1]
  \arrow["{\rm{refl}}"'{pos=0.3}, from=2-4, to=1-5]
  \arrow["{(\rm{loop}\ x)\inv}"'{pos=0.1}, from=4-2, to=5-1]
  \arrow["{\rm{refl}}"{pos=0.2}, from=4-4, to=5-5]
  \arrow["{\rm{loop}^{\rm{unglue}(w)}}"{marking, allow upside down}, draw=none, from=0, to=1]
\end{tikzcd}\]
~~~

```agda
  decode : ∀ x → Cover x → base ≡ x
  decode base (w , _) = loopʷ w
  decode (loop x i) w j = hcomp (∂ i ∨ ∂ j) λ where
    k (k = i0) → loopʷ (unglue w .fst) j
    k (i = i0) → ∙→square (loop-act (inl x) (w .fst)) (~ k) j
    k (i = i1) → loopʷ (w .fst) j
    k (j = i0) → base
    k (j = i1) → loop x (i ∨ ~ k)
```

To show that these functions are inverses at the base point, we do some

> more case analysis--- in one direction, this "case analysis" is path

Reviewer: ```suggestion
more case analysis --- in one direction, this "case analysis" is path
```

    ... | no ¬a = λ i →
        inl (transp (λ _ → A) i x)
      ∷ transp (λ _ → A ⊎ A) i y
      ∷ transp (λ _ → List (A ⊎ A)) i w

    encode-loopˡ (y ∷ w) r (inr x) with inr x ≡? opp y
    ... | yes a rewrite ≡?-yes' (a ∙ ap opp (sym (transport-refl y)))  =
      transport-refl _
    ... | no ¬a rewrite ≡?-no λ p → ¬a (p ∙ ap opp (transport-refl y)) = λ i →
      inr x ∷ transp (λ i → A ⊎ A) i y ∷ transp (λ i → List (A ⊎ A)) i w

    encode-loopʷ : (w : Word) (r : Reduced w) (w' : ⌞ Free-group ⌟) → ε (loopʷ w) w' .fst ≡ mul w (w' .fst)
    encode-loopʷ [] r w' = transport-refl (w' .fst)
    encode-loopʷ (x ∷ w) r w' =
      ε (loopʷ w ∙ loopˡ x) w'                                .fst ≡⟨ ap fst (subst-∙ Cover (loopʷ w) (loopˡ x) w') ⟩
      ε (loopˡ x) ⌜ ε (loopʷ w) w' ⌝                          .fst ≡⟨ ap! (Σ-prop-path! (encode-loopʷ w (uncons-reduced r) w')) ⟩
      ε (loopˡ x) (mul w (w' .fst) , mul-reduced w (w' .snd)) .fst ≡⟨ encode-loopˡ (mul w (w' .fst)) (mul-reduced w (w' .snd)) x ⟩
      act x (mul w (w' .fst))                                      ∎
```

</details>

As a consequence, the encoding of the word $w$ is precisely the product

> $w1$, which is of course just $w$--- this shows that encoding is inverse

Reviewer: ```suggestion
$w \cdot 1$, which is of course just $w$--- this shows that encoding is inverse
```

> $w1$, which is of course just $w$--- this shows that encoding is inverse

Reviewer: ```suggestion
$w1$, which is of course just $w$ --- this shows that encoding is inverse
```
